examples/common/utility/utility.hpp

```diff
@@ -356,6 +358,46 @@ class cli_argument_pack {
     }
 }; // class cli_argument_pack
 
+// utility class to aid relative error measurement of samples
+class measurements {
+public:
+    measurements() {
+        clear();
+    }
+    void clear() {
+        _secPerFrame.clear();
+    }
+    void start() {
+        _startTime = std::chrono::steady_clock::now();
+    }
+    void stop() {
+        _endTime = std::chrono::steady_clock::now();
+        std::chrono::duration<double> duration = _endTime - _startTime;
+        // store the duration in seconds
+        _secPerFrame.push_back(duration.count());
+    }
+    double computeRelError() {
+        auto averageTimePerFrame =
+            std::accumulate(_secPerFrame.begin(), _secPerFrame.end(), 0.0) / _secPerFrame.size();
+        std::vector<double> diff(_secPerFrame.size());
+        std::transform(_secPerFrame.begin(),
+                       _secPerFrame.end(),
+                       diff.begin(),
+                       [averageTimePerFrame](double x) {
+                           return (x - averageTimePerFrame);
+                       });
+        double sumOfSquareDiff = std::inner_product(diff.begin(), diff.end(), diff.begin(), 0.0);
+        double stdDev = std::sqrt(sumOfSquareDiff / _secPerFrame.size());
+        double relError = 100 * (stdDev / averageTimePerFrame);
+        return relError;
+    }
+
+private:
+    std::vector<double> _secPerFrame;
+    std::chrono::steady_clock::time_point _startTime;
+    std::chrono::steady_clock::time_point _endTime;
```

Comment:
To avoid having two vectors, where one is for `_startTime` timings and another one is for `_endTime` timings, and hence two separate memory pre-allocations, we may have single vector consisting of corresponding pairs of time points, that is:
```suggestion
    using time_point = std::chrono::steady_clock::time_point;
    time_point _startTime;
    std::vector< std::pair<time_point, time_point> > _time_intervals;
```
`_startTime` variable is constantly re-used during the `start()` call, and the `_time_intervals` being pre-allocated in the constructor, is pushed new pair into it during the `stop()` method call.

What do you think?